deltacat/compute/compactor_v2/steps/merge.py

```diff
@@ -306,13 +299,114 @@ def _compact_tables(
         compacted_table=compacted_table,
     )
     total_deduped_records = hb_table_record_count - len(table)
+    return table, incremental_len, total_deduped_records, merge_time
+
+
+def _compact_incremental_and_delete_table(
+    input: MergeInput,
+    df_envelopes: Optional[List[DeltaFileEnvelope]],
+    hb_idx: int,
+) -> Tuple[Any, int, int]:
+    table = None
+    total_incremental_len = 0
+    total_deduped_records = 0
+    total_dropped_records = 0
+    total_merge_time = 0
+    if not df_envelopes:
```

Comment:
We don't need to have this special case if we just create an holistic dfe list including deletes.